### PR TITLE
feat: confirm to remove signer from refused by list & added proposed by to census info

### PR DESCRIPTION
**Pull request by ifaouibadi**

### CODE REVIEW

**Review by marc0olo — APPROVED** [review]
I missed to cover that case, thanks!

lgtm

**Comment by marc0olo** [comment]
@ifaouibadi can you also update the README accordingly and add the new attribute for `consensus_info` ?

File: README.md [hunk 1]
```diff
@@ -84,6 +84,7 @@ These entrypoints are mainly for information purposes and required to build a me
         - `has_consensus` bool that indicates if the proposed tx was confirmed by a sufficient amount of signers
         - `expiration_height` the block height where the proposed tx expires
         - `expired` bool that indicates if the proposed tx is expired
+        - `proposed_by` refers to the address of the individual who initiated the proposal
```
Comment (thepiwo) [inline README.md:87]: bit wastefull to save it in state additionally if not used in contract logic

File: contracts/SimpleGAMultiSig.aes [hunk 2]
```diff
@@ -123,11 +124,16 @@ contract SimpleGAMultiSig =
     require_valid_signer_and_tx_hash(Call.caller, tx_hash, tx)
     require(!Set.member(Call.caller, tx.confirmed_by), "ERROR_ALREADY_CONFIRMED")
     require(!expired(tx), "ERROR_TX_EXPIRED")
-    let updated_tx = tx{ confirmed_by = Set.insert(Call.caller, tx.confirmed_by) }
-    put(state{ current_tx = Some(updated_tx) })
-    Chain.event(TxConfirmed(tx.tx_hash, Call.caller))
-    if(state.confirmations_required == Set.size(updated_tx.confirmed_by))
-      Chain.event(TxConsensusReached(tx.tx_hash))
+    //
```
Comment (thepiwo) [inline contracts/SimpleGAMultiSig.aes]: you wanted to comment here? :D

File: contracts/SimpleGAMultiSig.aes [hunk 3]
```diff
@@ -123,11 +124,16 @@ contract SimpleGAMultiSig =
...
+    switch(Set.member(Call.caller, tx.refused_by))
+      true =>
+        let updated_tx = tx{ refused_by = Set.delete(Call.caller, tx.refused_by),
```
Comment (thepiwo) [inline contracts/SimpleGAMultiSig.aes:129]: why is this needed? I guess it should be ok to have my address as confirmed and refused in order for others to make the decision, even if not needed it doesn't hurt and will be cheaper in gas

File: README.md (hunk 1 above)
Comment (marc0olo) [reply README.md:87]: we have that already stored in the state. Badi just added it to the consensus_info response. it is probably debatable but doesn't really hurt IMO

Comment (thepiwo) [reply README.md:87]: right, then its not a problem, my mistake

File: contracts/SimpleGAMultiSig.aes (hunk 3 above)
Comment (ifaouibadi) [reply contracts/SimpleGAMultiSig.aes:129]: we need it to determine what to display in the UI (accepted/rejected) cause now it's not possible to know that while the same address is in both `approvedBy` and `refusedBy`

File: README.md (hunk 1 above)
Comment (ifaouibadi) [reply README.md:87]: we had to make many requests to retrieve the person, who created the proposal
I think it should come along with the census info

File: contracts/SimpleGAMultiSig.aes (hunk 3 above)
Comment (thepiwo) [reply contracts/SimpleGAMultiSig.aes:129]: can't both be shown?

Comment (thepiwo) [reply contracts/SimpleGAMultiSig.aes:129]: I think we should build the UI to serve the contract, not the contract to serve the UI

Comment (marc0olo) [reply contracts/SimpleGAMultiSig.aes:129]: I think it is more a logical UX thing. users would be very confused to see a signer in both lists. it just doesn't make sense from user perspective

Comment (thepiwo) [reply contracts/SimpleGAMultiSig.aes:129]: as discussed with @ifaouibadi it will be important to revoke accidental approval, so this change is different logic but probably better.

**Review by thepiwo — APPROVED** [review]
